File: web/src/app/model/ticket-attribute.ts
import {Attribute} from "./attribute";

export class TicketAttribute {
    attribute: Attribute;

    value: string;

    color: string;

    static from(value: any): TicketAttribute {
        let ticketAttribute = new TicketAttribute();
        ticketAttribute.attribute = Attribute.from(value.attribute);
        ticketAttribute.value = value.value;
        ticketAttribute.color = value.color;
        return ticketAttribute;
    }

    toSave(): any {
        return {
            attribute: this.attribute.toSave(),
            value: this.value
        }
    }
}

File: server/src/main/java/com/evgenltd/lt/entity/TicketAttribute.java
package com.evgenltd.lt.entity;

import javax.persistence.*;

@Entity
@Table(name = "ticket_attributes")
public class TicketAttribute {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long id;

    @ManyToOne
    @JoinColumn(name = "ticket_id")
    private Ticket ticket;

    @ManyToOne
    @JoinColumn(name = "attribute_id")
    private Attribute attribute;

    private String value;

    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }

    public Ticket getTicket() {
        return ticket;
    }

    public void setTicket(final Ticket ticket) {
        this.ticket = ticket;
    }

    public Attribute getAttribute() {
        return attribute;
    }

    public void setAttribute(final Attribute attribute) {
        this.attribute = attribute;
    }

    public String getValue() {
        return value;
    }

    public void setValue(final String value) {
        this.value = value;
    }
}
